airflow/providers/amazon/aws/transfers/redshift_to_s3.py

```diff
@@ -143,8 +144,17 @@ def __init__(
     def _build_unload_query(
         self, credentials_block: str, select_query: str, s3_key: str, unload_options: str
     ) -> str:
+        # TODO: remove in provider 8.0.0
+        if "''" in select_query:
+            logging.warning(
```

Comment:
Why do you want to remove this check at the next major release? I dont understand. Users, at any point in time, can use this operator and specify a query with already double quoted strings

Reply:
+1
Users may stumble on this in the future after you have removed the check.

Also is it possible the string contains some double quotes but not everywhere? Do we want to try handle these half working situations? This type of string sanitization code can often get very hairy very quick. Is there a thirdparty library we can use instead which specializes in this?

Reply:
I dont know any library doing that but that'd be cool. Otherwise, I am in favor of NOT managing these kind of use cases. These use cases (mix of quotes escaped and some unescaped) are not valid and users should fix it. The current implementation would not alter the string which, I think, is the correct behavior